soccer/motion/MotionControl.cpp

```diff
@@ -35,244 +30,150 @@ void MotionControl::createConfiguration(Configuration* cfg) {
+    if (maybe_target) {
+        auto target = maybe_target.value();
+        maybe_pose_target = target.pose();
+        velocity_target = target.twist();
 
-        return;  // pivot handles both angle and position
+        // If we have no goal angle, set it to the current angle.
+        if (!(target.angle && target.angle.value().angle)) {
+            maybe_pose_target.value().heading() = state.pose.heading();
+        }
     }
-     */
-
-    // Position control ///////////////////////////////////////////////
-    ////////////////////////////////////////////////////////////////////
-    MotionInstant target = optTarget->motion;
 
-    // tracking error
-    Point posError = target.pos - _robot->pos();
+    // TODO: Calculate acceleration and use it to improve response.
+    // TODO: Clamp acceleration
 
-    // acceleration factor
-    Point acceleration;
-    std::optional<RobotInstant> nextTarget =
-        _robot->path().evaluate(timeIntoPath + RJ::Seconds(1) / 60.0);
-    if (nextTarget) {
-        acceleration = (nextTarget->motion.vel - target.vel) / 60.0f;
+    Twist correction = Twist::Zero();
+    if (maybe_pose_target) {
+        Pose error = maybe_pose_target.value() - state.pose;
+        error.heading() = fixAngleRadians(error.heading());
+        correction = Twist(_positionXController.run(error.position().x()),
+                           _positionYController.run(error.position().x()),
+                           _angleController.run(error.heading()));
     } else {
-        acceleration = {0, 0};
+        reset();
     }
-    Point accelFactor =
-        acceleration * 60.0f * (*_robot->config->accelerationMultiplier);
 
-    target.vel += accelFactor;
+    // Apply the correction and rotate into the world frame.
+    Twist result_world = velocity_target + correction;
+    Twist result_body(
+        result_world.linear().rotated(M_PI_2 - state.pose.heading()),
+        result_world.angular());
 
-    // PID on position
-    target.vel.x() += _positionXController.run(posError.x());
-    target.vel.y() += _positionYController.run(posError.y());
+    // Use default constraints. Planning should be in charge of enforcing
+    // constraints on the path, here we just follow it.
+    // TODO(Kyle): Use this robot's constraints here.
+    RobotConstraints constraints;
 
-    // draw target pt
-    _context->debug_drawer.drawCircle(target.pos, .04, Qt::red,
-                                      "MotionControl");
-    _context->debug_drawer.drawLine(target.pos, target.pos + target.vel,
-                                    Qt::blue, "MotionControl");
-
-    // Clamp World Acceleration
-    auto dt = RJ::Seconds(RJ::now() - _lastCmdTime);
-    Point targetAccel = (target.vel - _lastWorldVelCmd) / dt.count();
-    targetAccel.clamp(*_max_acceleration);
-
-    target.vel = _lastWorldVelCmd + targetAccel * dt.count();
-
-    _lastWorldVelCmd = target.vel;
-    _lastCmdTime = RJ::now();
```

Comment:
Just to throw this out there, it's more correct without all the motion control hacks, so removing them is the "correct" thing to do, but it did bite us in the ass when I did that on firmware.  It might be prudent to leave them in until tuning phase, and then remove them if they're not needed.

Reply:
That's true. However, the minimum-velocity hacks actually significantly (negatively) impact sim performance, and so I think they belong on the robot side of motion control regardless. I'd honestly rather optimize this part for sim anyways, considering that we are who knows how far from seeing an actual robot, and the first thing we do with new robots is motion control work where I'd rather have a clean(er) slate to work with anyway.

Acceleration _feedthrough_ is something that I think probably does belong here. I'll add it in tomorrow. Acceleration limiting is more questionable  - there's an argument to be said that it belongs on the robot, or that it belongs here, but either way I don't think it will be making much difference considering the motion planning already acceleration limits the goal (so it only matters here if feedback is large).